### ISSUE TITLE
Message CompletionStage is never aborted in case of error

### ISSUE THREAD

**Issue by andreas-eberle** [smallrye/smallrye-reactive-messaging#487]
## Description

When you try to send a message with an `Emitter`, you either get a `CompletionStage` from the `Emitter.send(T)` method  or you can register one on the `Message` object. Doing this seems to be the only way to be notified when the message has actually been sent. 

However, if the message sending fails due to any reason (e.g. the kafka broker is temporarily down), the `CompletionStage` will never be completed or aborted, leaving anyone waiting for the message sending waiting there forever.

Currently, I cannot see any way of knowing that a message acutally has been sent (which is important for my application) without risiking to have never closing connections. 

Is there any way to reliably find out if a message was sent successfully?

Background: I have a rest call, that needs to create a message and must only return successful, if the message was created or fail if it couldn't be created.

## Reproducer
1. Download [reactive-messaging-no-error.zip](https://github.com/smallrye/smallrye-reactive-messaging/files/4450971/reactive-messaging-no-error.zip) and unzip it.
2. Simply start the project (`./mvnw quarkus:dev`) without having a kafka running
3. Call the `ExampleResource` by opening http://localhost:8080/hello
4. The log shows `hello() called` showing that the resource was called but the `CompletionStage` never finishes.
5. Start a kafka, call the resource again and see that it works.

**Comment by cescoffier** [comment]
Thanks for the report.

The callback is invoked when the message get acknowledged. If the message is not getting acknowledged, we should fail the completion stage (`completeExceptionally`).

**Comment by zeljkot** [comment]
This is how I handle it for multiple messages. For a single message it could be even simpler because you already have Uni, which is needed for `.ifNoItem`.

```java
.emitOn(Infrastructure.getDefaultWorkerPool())
.onItem().produceCompletionStage(fileEntity -> sendMessage(fileEntity)).concatenate()
.collectItems().asList()
.ifNoItem().after(Duration.ofSeconds(60)).failWith(
    () -> new ServerErrorException("Kafka not responding", Response.Status.GATEWAY_TIMEOUT));
```

**Comment by cescoffier** [comment]
Fixed in #573
